gcloud-java-logging/src/main/java/com/google/gcloud/logging/spi/v2/ConfigServiceV2Api.java

```diff
@@ -0,0 +1,533 @@
+   * <!-- end manual edit -->
+   */
+  public static ConfigServiceV2Api create(ServiceApiSettings settings) throws IOException {
+    return new ConfigServiceV2Api(settings);
+  }
+
+  /**
+   * Constructs an instance of ConfigServiceV2Api, using the given settings. This is protected so that it
+   * easy to make a subclass, but otherwise, the static factory methods should be preferred.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   */
+  protected ConfigServiceV2Api(ServiceApiSettings settings) throws IOException {
+    ServiceApiSettings internalSettings = ApiUtils.populateSettings(settings,
+        SERVICE_ADDRESS, DEFAULT_SERVICE_PORT, ALL_SCOPES);
+    this.settings = internalSettings;
+    this.channel = internalSettings.getChannel();
+  }
+
+  // ==============================
+  // Resource Name Helper Functions
+  // ==============================
+
+  /**
+   * Creates a string containing the fully-qualified path to represent
+   * a project_name resource.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   */
+  public static final String createProjectNamePath(String project) {
+    return PROJECT_NAME_PATH_TEMPLATE.instantiate(
+        "project", project);
+  }
+
+  /**
+   * Creates a string containing the fully-qualified path to represent
+   * a sink_name resource.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   */
+  public static final String createSinkNamePath(String project, String sink) {
+    return SINK_NAME_PATH_TEMPLATE.instantiate(
+        "project", project,"sink", sink);
+  }
+
+
+  /**
+   * Extracts the project from the given fully-qualified path which
+   * represents a projectName resource.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   */
+  public static final String extractProjectFromProjectNamePath(String projectNamePath) {
+    return PROJECT_NAME_PATH_TEMPLATE.parse(projectNamePath).get("project");
+  }
+
+  /**
+   * Extracts the project from the given fully-qualified path which
+   * represents a sinkName resource.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   */
+  public static final String extractProjectFromSinkNamePath(String sinkNamePath) {
+    return SINK_NAME_PATH_TEMPLATE.parse(sinkNamePath).get("project");
+  }
+
+  /**
+   * Extracts the sink from the given fully-qualified path which
+   * represents a sinkName resource.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   */
+  public static final String extractSinkFromSinkNamePath(String sinkNamePath) {
+    return SINK_NAME_PATH_TEMPLATE.parse(sinkNamePath).get("sink");
+  }
+
+
+  // =============
+  // Service Calls
+  // =============
+
+  // ----- listSinks -----
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Lists sinks.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   */
+  public Iterable<LogSink> listSinks(String projectName) {
```

Comment:
If method is idempotent, the client library will take care of retrying calls for you.
As for the throws tag, #477 .